control-plane-operator/controllers/hostedcontrolplane/kas/params.go

```diff
@@ -82,6 +87,7 @@ func NewKubeAPIServerParams(ctx context.Context, hcp *hyperv1.HostedControlPlane
 		ServiceCIDR:          hcp.Spec.ServiceCIDR,
 		PodCIDR:              hcp.Spec.PodCIDR,
 		Availability:         hcp.Spec.ControllerAvailabilityPolicy,
+		ConsolePublicURL:     fmt.Sprintf("console-openshift-console.%s", dns.Spec.BaseDomain),
```

Comment:
where is this used?

Reply:
That’s used to properly perform the redirect when a user access the kube master endpoint in a browser as unauthenticated.

effectively it will then redirect the user to the console url I’ll give an example with pictures

Reply:
But where is it set in the KAS config?

Reply:
Workflow:
1) Users will typically query a given master endpoint with
```
Tylers-MacBook-Pro:armada-hypershift-operator tylerlisowski$ bx cs cluster get --cluster bsr0toow08i9rajtg1a0
Retrieving cluster bsr0toow08i9rajtg1a0...
OK
                                   
Name:                           pvg-metrics-collector   
ID:                             bsr0toow08i9rajtg1a0   
State:                          normal   
Status:                         All Workers Normal   
Created:                        2020-08-13 23:11:15 -0500 (2 years ago)   
Resource Group ID:              f4bf7759b43c4d1a859e328dc363e478   
Resource Group Name:            default   
Pod Subnet:                     172.17.64.0/18   
Service Subnet:                 172.21.0.0/16   
Workers:                        3   
Worker Zones:                   us-east-1   
Ingress Subdomain:              pvg-metrics-collector-80d128fecd199542426020c17e5e9430-0000.us-east.containers.appdomain.cloud   
Ingress Secret:                 pvg-metrics-collector-80d128fecd199542426020c17e5e9430-0000   
Ingress Status:                 healthy   
Ingress Message:                All Ingress components are healthy   
Public Service Endpoint URL:    https://c100-e.us-east.containers.cloud.ibm.com:31581   
Private Service Endpoint URL:   https://c100.private.us-east.containers.cloud.ibm.com:31581   
Pull Secrets:                   enabled in the default namespace   
VPCs:                           r014-d56f656e-96d4-4a7e-bd7d-7dc8cfa2eaa7   

Master         
Status:     Ready (5 days ago)   
State:      deployed   
Health:     normal   
Version:    4.9.21_1528_openshift   
Location:   Washington D.C.   
URL:        https://c100-e.us-east.containers.cloud.ibm.com:31581   
Tylers-MacBook-Pro:armada-hypershift-operator tylerlisowski$ 
```
![Screen Shot 2022-03-09 at 8 51 17 AM](https://user-images.githubusercontent.com/7716924/157466061-42913e0e-4e0b-467a-9a12-ce9c48090dcd.png)


2) They can then paste that master endpoint into the browser

Reply:
When doing so they get redirected to the console:
![Screen Shot 2022-03-09 at 8 51 37 AM](https://user-images.githubusercontent.com/7716924/157466171-987fc2d2-98ab-451f-98ac-c14c2ec7745f.png)

Reply:
Without this that does not work: showing an example in hypershift
```
Retrieving cluster c8ig44h10gqakshf6pfg...
OK
                                   
Name:                           tyler-hyp-20220306-1-d1   
ID:                             c8ig44h10gqakshf6pfg   
State:                          normal   
Status:                         All Workers Normal   
Created:                        2022-03-06 12:48:50 -0600 (2 days ago)   
Resource Group ID:              c0b7468bdb034851964ec1b3a0969fd8   
Resource Group Name:            Default   
Pod Subnet:                     172.30.0.0/16   
Service Subnet:                 172.21.0.0/16   
Workers:                        2   
Worker Zones:                   us-south-1   
Ingress Subdomain:              tyler-hyp-20220306-1-9e37478581b5d9de33607f5926d1d18f-0000.upi.prestg.stg.containers.appdomain.cloud   
Ingress Secret:                 tyler-hyp-20220306-1-9e37478581b5d9de33607f5926d1d18f-0000   
Ingress Status:                 -   
Ingress Message:                -   
Public Service Endpoint URL:    https://c-01.private.us-south.link.satellite.test.cloud.ibm.com:33572   
Private Service Endpoint URL:   https://t36c414f51590ecd4fb40-d603ff82e51c94176a53d44566df9d79-ce00.us-south.satellite.test.appdomain.cloud:32385   
Pull Secrets:                   enabled in the default namespace   
VPCs:                           -   

Master         
Status:     Ready (2 days ago)   
State:      deployed   
Health:     normal   
Version:    4.9.22_1143_openshift   
Location:   tyler-hyp-20220306-1   
URL:        https://t36c414f51590ecd4fb40-d603ff82e51c94176a53d44566df9d79-ce00.us-south.satellite.test.appdomain.cloud:32385   
```
![Screen Shot 2022-03-09 at 8 54 47 AM](https://user-images.githubusercontent.com/7716924/157466741-a9ceea44-7e4b-4b76-88de-84bd9da13396.png)

Reply:
yea great point just wanted to make sure you were cool with general idea: updated!